packages/tre/src/plugins/cache/errors.ts

```diff
@@ -0,0 +1,69 @@
+import { Response } from "undici";
+import { CfHeader } from "../shared/constants";
+
+enum Status {
+  PayloadTooLarge = 413,
+  NotFound = 404,
+  CacheMiss = 504,
+}
+
+export async function fallible<T>(promise: Promise<T>): Promise<T | Response> {
+  try {
+    return await promise;
+  } catch (e) {
+    if (e instanceof CacheError) {
+      return e.toResponse();
+    }
+    throw e;
+  }
+}
+
+export class CacheError extends Error {
```

Comment:
I copied this over from `R2Error`, but the reason it uses `status` rather than `code` is because it is the http _status_ code. `R2Error` has another property `v4Code`, so using code there would be a bit confusing, I think. For consistency, it might make sense to keep this as `status`. I'll have a look at extending from `MiniflareError` though